src/packagist.js

```diff
@@ -0,0 +1,93 @@
+const https = require('https');
+
+const packagesCache = {};
+
+/**
+ * Packagist Search API
+ *
+ * Response format will be like:
+ * {
+ *  "results":[
+ *    {
+ *      "name":"mage-os/magento-cache-clean",
+ *      "description":"A faster drop in replacement for bin/magento cache:clean with file watcher",
+ *      "url":"https://packagist.org/packages/mage-os/magento-cache-clean",
+ *      "repository":"https://github.com/mage-os/magento-cache-clean",
+ *      "downloads":44773,
+ *      "favers":29
+ *    }
+ *  ],
+ *  "total":23,
+ *  "next": "https://packagist.org/search.json?q=[query]&page=[next page number]"
+ * }
+ *
+ * @see https://packagist.org/apidoc#search-packages
+ */
+
+/**
+ * @param {string} url 
+ * @returns {Promise<any>}
+ */
+function fetchUrl(url) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      let data = '';
+      res.on('data', chunk => data += chunk);
+      res.on('end', () => {
+        try {
+          resolve(JSON.parse(data));
+        } catch (err) {
+          reject(err);
+        }
+      });
+    }).on('error', reject);
+  });
+}
+
+/**
+ * Fetch all packages for a vendor from Packagist, handling pagination.
+ * @param {string} vendor
+ * @returns {Promise<Set<string>>}
+ */
+async function fetchPackagistList(vendor) {
+  if (packagesCache[vendor]) {
+    return packagesCache[vendor];
+  }
+
+  let url = 'https://packagist.org/search.json?per_page=100&q=' + encodeURIComponent(vendor + '/');
```

Comment:
Is 100 the allowed maximum?
If not, it may be better to use a larger number, i.e. 1000.
I'm afraid we will bump into the limit and not notice.
I guess if 100 is the max. pagination needs to be added - later ;)

Reply:
I believe 100 is the max, but it's not spelled out clearly. I didn't test that to verify. It will automatically fetch multiple pages if there's more than one (at least, it should) via the while loop on `json.next` just below:

https://github.com/mage-os/generate-mirror-repo-js/pull/235/files/64101b7fae12625c1876b02b06e66a0a3981f426#diff-7afa5fd41fecaa24012ced417c756d4efe88f93f34a6c4d201637abed6db46d4R60-R66

API is documented at https://packagist.org/apidoc#search-packages